app/templates/components/public/side-menu.hbs

```diff
@@ -1,25 +1,37 @@
 <div class="ui fluid vertical {{unless device.isMobile 'pointing'}} menu">
   {{#if (eq session.currentRouteName 'public.index')}}
-    {{#scroll-to href='#info' class='item active'}}
-      {{t 'Info'}}
-    {{/scroll-to}}
-    {{#scroll-to href='#tickets' class='item'}}
-      {{t 'Tickets'}}
-    {{/scroll-to}}
-    {{#scroll-to href='#speakers' class='item'}}
-      {{t 'Speakers'}}
-    {{/scroll-to}}
+    {{#if event.description}}
+      {{#scroll-to href='#info' class='item active'}}
+        {{t 'Info'}}
+      {{/scroll-to}}
+    {{/if}}
+    {{#if event.tickets}}
+      {{#scroll-to href='#tickets' class='item'}}
+        {{t 'Tickets'}}
+      {{/scroll-to}}
+    {{/if}}
+    {{#if event.speakers}}
+      {{#scroll-to href='#speakers' class='item'}}
+        {{t 'Speakers'}}
+      {{/scroll-to}}
+    {{/if}}
   {{else}}
+  {{#if event.description}}
     <a class="item" href="{{href-to 'public.index'}}">
       {{t 'Info'}}
     </a>
+  {{/if}}
+  {{#if event.tickets}}
     <a class="item" href="{{href-to 'public.index'}}#tickets">
       {{t 'Tickets'}}
     </a>
+  {{/if}}
+  {{#if event.speakers}}
     <a class="item" href="{{href-to 'public.index'}}#speakers">
       {{t 'Speakers'}}
     </a>
   {{/if}}
+  {{/if}}
   {{#link-to 'public.sessions' class='item'}}
```

Comment:
@niranjan94 The thing here is when we try to do this, the acceptance test for sessions fails. Since in the test we are visiting the public/index route and clicking on the sessions navigation item. But since it is not there, it fails.
```
test('visiting / and opening an event and going to sessions', function(assert) {
  visit('/');
  andThen(function() {
    click('div.event.card:first > a.content:first');
    andThen(function() {
      assert.equal(currentRouteName(), 'public.index');
      click('div.ui.vertical.pointing.menu > a:contains(\'Sessions\')');
      andThen(function() {
        assert.equal(currentRouteName(), 'public.sessions');
      });
    });
  });
});
```
I tried using something like:
```
test('visiting / and opening an event and going to sessions', function(assert) {
  visit('/');
  andThen(function() {
    click('div.event.card:first > a.content:first');
    andThen(function() {
      assert.equal(currentRouteName(), 'public.index');
      if (find('div.ui.vertical.pointing.menu > a:contains(\'Sessions\')')) {
        click('div.ui.vertical.pointing.menu > a:contains(\'Sessions\')');
        andThen(function() {
          assert.equal(currentRouteName(), 'public.sessions');
        });        
      }
    });
  });
});
```
But this does not help.